website/docs/r/vpc_gateway_policy.html.markdown

```diff
@@ -0,0 +1,123 @@
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+```
+
+-> We recommend using `lifecycle` directive as in samples above, in order to avoid dependency issues when updating groups/services simultaneously with the rule.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `display_name` - (Required) Display name of the resource.
+* `category` - (Required) The category to use for priority of this Gateway Policy. For local manager must be one of: `Emergency`, `SystemRules`, `SharedPreRules`, `LocalGatewayRules`, `AutoServiceRules` and `Default`. For global manager must be `SharedPreRules` or `LocalGatewayRules`.
+* `description` - (Optional) Description of the resource.
+* `tag` - (Optional) A list of scope + tag pairs to associate with this Gateway Policy.
+* `nsx_id` - (Optional) The NSX ID of this resource. If set, this ID will be used to create the Gateway Policy resource.
+* `context` - (Required) The context which the object belongs to
+  * `project_id` - (Required) The ID of the project which the object belongs to
+  * `vpc_id` - (Required) The ID of the VPC which the object belongs to
+* `comments` - (Optional) Comments for this Gateway Policy including lock/unlock comments.
+* `locked` - (Optional) A boolean value indicating if the policy is locked. If locked, no other users can update the resource.
+* `sequence_number` - (Optional) An int value used to resolve conflicts between security policies
+* `stateful` - (Optional) A boolean value to indicate if this Policy is stateful. When it is stateful, the state of the network connects are tracked and a stateful packet inspection is performed.
+* `tcp_strict` - (Optional) A boolean value to enable/disable a 3 way TCP handshake is done before the data packets are sent.
+* `rule` (Optional) A repeatable block to specify rules for the Gateway Policy. Each rule includes the following fields:
+  * `display_name` - (Required) Display name of the resource.
+  * `description` - (Optional) Description of the resource.
+  * `destination_groups` - (Optional) Set of group paths that serve as the destination for this rule. IPs, IP ranges, or CIDRs may also be used starting in NSX-T 3.0. An empty set can be used to specify "Any".
+  * `destinations_excluded` - (Optional) A boolean value indicating negation of destination groups.
+  * `direction` - (Optional) The traffic direction for the policy. Must be one of: `IN`, `OUT` or `IN_OUT`. Defaults to `IN_OUT`.
+  * `disabled` - (Optional) A boolean value to indicate the rule is disabled. Defaults to `false`.
+  * `ip_version` - (Optional) The IP Protocol for the rule. Must be one of: `IPV4`, `IPV6` or `IPV4_IPV6`. Defaults to `IPV4_IPV6`.
+  * `logged` - (Optional) A boolean flag to enable packet logging.
+  * `notes` - (Optional) Text for additional notes on changes for the rule.
+  * `profiles` - (Optional) A list of context profiles for the rule. Note: due to platform issue, this setting is only supported with NSX 3.2 onwards.
```

Comment:
Lets remove NSX version mention here as well

Reply:
FQDN network policies should work for VPCs as well.